From 3dfd3a2b64592905be8e7bd6d229e0f7d077ed54 Mon Sep 17 00:00:00 2001
From: Jason King <jason.king@joyent.com>
Date: Mon, 16 Jul 2018 18:37:54 +0000
Subject: [PATCH] OS-7079 mp_startup_common races itself Reviewed by: Robert
 Mustacchi <rm@joyent.com> Reviewed by: Patrick Mooney
 <patrick.mooney@joyent.com> Approved by: Patrick Mooney
 <patrick.mooney@joyent.com>

---
 usr/src/uts/i86pc/os/mp_startup.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/usr/src/uts/i86pc/os/mp_startup.c b/usr/src/uts/i86pc/os/mp_startup.c
index a881ddc3c5..fef8f2759d 100644
--- a/usr/src/uts/i86pc/os/mp_startup.c
+++ b/usr/src/uts/i86pc/os/mp_startup.c
@@ -1900,6 +1900,8 @@ mp_startup_common(boolean_t boot)
 	if (boothowto & RB_DEBUG)
 		kdi_cpu_init();
 
+	(void) mach_cpu_create_device_node(cp, NULL);
+
 	/*
 	 * Setting the bit in cpu_ready_set must be the last operation in
 	 * processor initialization; the boot CPU will continue to boot once
@@ -1907,8 +1909,6 @@ mp_startup_common(boolean_t boot)
 	 */
 	CPUSET_ATOMIC_ADD(cpu_ready_set, cp->cpu_id);
 
-	(void) mach_cpu_create_device_node(cp, NULL);
-
 	cmn_err(CE_CONT, "?cpu%d: %s\n", cp->cpu_id, cp->cpu_idstr);
 	cmn_err(CE_CONT, "?cpu%d: %s\n", cp->cpu_id, cp->cpu_brandstr);
 	cmn_err(CE_CONT, "?cpu%d initialization complete - online\n",
-- 
2.21.0

